src/main/scala/org/broadinstitute/dsde/firecloud/model/Attributes.scala

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.dsde.firecloud.model
+
+import org.broadinstitute.dsde.firecloud.FireCloudException
+
+object Attributable {
+  type AttributeMap = Map[AttributeName, Attribute]
+}
+
+case class AttributeName(
+                          namespace: String,
+                          name: String) extends Ordered[AttributeName] {
+  // enable implicit ordering for sorting
+  import scala.math.Ordered.orderingToOrdered
+  def compare(that: AttributeName): Int = (this.namespace, this.name) compare (that.namespace, that.name)
+}
+
+object AttributeName {
+  val defaultNamespace = "default"
+  val libraryNamespace = "library"
+  val validNamespaces = Set(AttributeName.defaultNamespace, AttributeName.libraryNamespace)
+
+  val delimiter = ':'
+
+  def withDefaultNS(name: String) = AttributeName(defaultNamespace, name)
+
+  def toDelimitedName(aName: AttributeName): String = {
+    if (aName.namespace == defaultNamespace) aName.name
+    else aName.namespace + delimiter + aName.name
+  }
+
+  def fromDelimitedName(dName: String): AttributeName = {
+    dName.split(delimiter).toList match {
+      case sName :: Nil => AttributeName.withDefaultNS(sName)
+      case sNamespace :: sName :: Nil => AttributeName(sNamespace, sName)
+      case _ => throw new FireCloudException(s"Attribute string $dName has too many '$delimiter' delimiters")
+    }
+  }
+}
+
+sealed trait Attribute
+sealed trait AttributeValue extends Attribute
+sealed trait AttributeList[T <: Attribute] extends Attribute { val list: Seq[T] }
```

Comment:
I realize you copied this from rawls, but ... should this be: `AttributeList[T <: AttributeValue]`?

Otherwise, we technically support lists-of-lists. Do we do that? Do we need T <: Attribute in order to support AttributeEntityReference, and if so is there another way to do it where we *don't* support lists-of-lists?